stdlib/test/memory/test_memory.mojo

```diff
@@ -518,6 +518,25 @@ def test_indexing():
     assert_equal(ptr[1], 1)
 
 
+def test_wide_load():
+    var ptr = DTypePointer[DType.bool].alloc(4)
+    ptr.store(0, True)
+    ptr.store(1, True)
+    ptr.store(2, True)
+    ptr.store(3, True)
+    assert_equal(
+        ptr.load[width=4](0), SIMD[DType.bool, 4](True, True, True, True)
+    )
+
+    var ptr2 = DTypePointer[DType.int32].alloc(4)
+    ptr2.store(0, 1)
+    ptr2.store(1, 2)
```

Comment:
Could you also add the inverse of this test, where you `store[width=4]()` and `load[width=1]()`?

Reply:
Good catch, `store[width=4]()` also doesn't work currently.